pkg/controller/node/node_controller.go

```diff
@@ -464,7 +464,9 @@ func (ctrl *Controller) updateNode(old, cur interface{}) {
 		return
 	}
 
-	pools, err := ctrl.getPoolsForNode(curNode)
+	// enqueue all pools from the "old" node object as this update call might
+	// include a change where a role has been dropped.
+	pools, err := ctrl.getPoolsForNode(oldNode)
```

Comment:
Sorry this is slightly confusing, wouldn't this ONLY sync pools for `oldNode`? In the proposed scenario it would work since oldNode has everything in curNode + more, but wouldn't the opposite be true (and cause problems) if you're trying to add more labels (more pools) to a node?

`pools` should be a merge of the two lists instead, right?